### PR TITLE
util: add check delete json function

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref #4399

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/utils/testutil/api_check.go [hunk 1]
```diff
@@ -123,9 +123,18 @@ func CheckPatchJSON(client *http.Client, url string, data []byte, checkOpts ...f
...
+	if err != nil {
+		return err
+	}
+	return checkResp(resp, checkOpts...)
+}
+
 func checkResp(resp *http.Response, checkOpts ...func([]byte, int, http.Header)) error {
 	res, err := io.ReadAll(resp.Body)
-	resp.Body.Close()
+	defer resp.Body.Close()
```
Comment (rleungx) [inline pkg/utils/testutil/api_check.go:137]: seem no need to change?

Comment (lhy1024) [reply pkg/utils/testutil/api_check.go:137]: It seems so.

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by rleungx — APPROVED** [review]

**Pull request merged**